### PR TITLE
[GEOT-6476] MongoDb store wrongly return false when supports sorting …

### PR DESCRIPTION
**Pull request by taba90**
…method is issued

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by taba90** [comment]
JIRA ticket https://osgeo-org.atlassian.net/browse/GEOT-6476

File: modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    private MongoFeatureSource source;
+
+    MongoQueryCapabilities(MongoFeatureSource source) {
+        this.source = source;
+    }
+
+    @Override
+    public boolean supportsSorting(SortBy[] sortAttributes) {
+        if (super.supportsSorting(sortAttributes)) return true;
```
Comment (nmco) [inline modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java:37]: Not sure about this one @taba90, even if the default implementation supports it we still want to override the default check no?

File: modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Check if the property name corresponds to one of the FeatureType attributes.
+     *
+     * @param propertyName
+     * @return true if a correspondence is found, false if not or the attribute is a Geometry.
+     */
+    private boolean supportsPropertySorting(PropertyName propertyName) {
+        AttributeDescriptor descriptor =
+                (AttributeDescriptor) propertyName.evaluate(source.getSchema());
+        if (descriptor == null) {
```
Comment (nmco) [inline modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java:53]: This would deserve a comment explaining why the descriptor may be NULL.

File: modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoDataStoreTest.java [hunk 3]
```diff
@@ -198,4 +206,27 @@ private void clearSchemaStore(MongoDataStore mongoStore) {
         }
         mongoStore.cleanEntries();
     }
+
+    public void testSortBy() throws Exception {
```
Comment (nmco) [inline modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoDataStoreTest.java:210]: This is an online test right? I would also add a test for DESCENDING.

Comment (nmco) [reply modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoDataStoreTest.java:210]: Forget about this just notice that you have that test on ``MongoFeatureSourceTest``.

File: modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java [hunk 4]
```diff
@@ -315,6 +320,62 @@ public void testAndPostFilter() throws Exception {
...
+        }
+        assertEquals(doubleValues.size(), 3);
+        Double first = doubleValues.get(0);
+        Double second = doubleValues.get(1);
+        Double third = doubleValues.get(2);
+        assertTrue(first > second);
+        assertTrue(second > third);
+    }
+
+    public void testTwoSortBy() throws Exception {
```
Comment (nmco) [inline modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java:350]: I would add a test or one more attribute, where one doesn't override it ignores the NULL, sorte by the next attribute and then by the next one, makes sense?

File: modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java (hunk 1 above)
Comment (taba90) [reply modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java]: fixed

File: modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java (hunk 2 above)
Comment (taba90) [reply modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoQueryCapabilities.java]: I removed that part, seems to be redundat cause PropertyName will use propertyAccessors one of which access the property in that exact way.

File: modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java (hunk 4 above)
Comment (taba90) [reply modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoFeatureSourceTest.java:350]: fixed

**Comment by taba90** [comment]
thnaks for your review @nmco your feedbacks have been applied

**Comment by nmco** [comment]
Thank you @taba90, looks good to me.
